src/background/background.js

```diff
@@ -734,6 +734,18 @@ async function copyMarkdownFromContext(info, tab) {
     else if (info.menuItemId == "copy-markdown-image") {
       await browser.tabs.executeScript(tab.id, {code: `copyToClipboard("![](${info.srcUrl})")`});
     }
+    else if(info.menuItemId == "copy-markdown-obsidian") {
+      const article = await getArticleFromContent(tab.id, info.menuItemId == "copy-markdown-obsidian");
+      const { markdown } = await convertArticleToMarkdown(article, downloadImages = false);
+      await browser.tabs.executeScript(tab.id, { code: `copyToClipboard(${JSON.stringify(markdown)})` });
+      document.location.href = "obsidian://advanced-uri?vault=&clipboard=true&mode=new&filepath=Clipper-" + generateValidFileName(tab.title)
+    }
+    else if(info.menuItemId == "copy-markdown-obsall") {
+      const article = await getArticleFromContent(tab.id, info.menuItemId == "copy-markdown-obsall");
+      const { markdown } = await convertArticleToMarkdown(article, downloadImages = false);
+      await browser.tabs.executeScript(tab.id, { code: `copyToClipboard(${JSON.stringify(markdown)})` });
+      document.location.href = "obsidian://advanced-uri?vault=&clipboard=true&mode=new&filepath=Clipper-" + generateValidFileName(tab.title)
```

Comment:
I'm not sure about prepending `Clipper-` to the filename. This won't be to everyone's tastes and if you wanted that personally, you could always add it to the template file title.

I'd also suggest getting the title like this (instead of just the tab title):
```js
  const article = await getArticleFromContent(tab.id, info.menuItemId == "download-markdown-selection");
  const title = await formatTitle(article);
  ```
... and prepending the `mdClips` folder, so the users can specify a subfolder of their vault they want the clip to go to

Reply:
Okay I will try to change the code. The `Clipper-` is just how I identified the clipper content for obsidian because I'm still new to browser extension. Maybe when I'm able to add new option for Folder name in setting page, I will adapt to that.

Reply:
Done in new commit